native/dlarfx.go

```diff
@@ -0,0 +1,480 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import "github.com/gonum/blas"
+
+// Dlarfx applies an elementary reflector H to a real m×n matrix C, from either
+// the left or the right, with loop unrolling when the reflector has order less
+// than 11.
+//
+// H is represented in the form
+//  H = I - tau * v * v^T,
+// where tau is a real scalar and v is a real vector. If tau = 0, then H is
+// taken to be the identity matrix.
+//
+// v must have length equal to m if side == blas.Left, and equal to n if side ==
+// blas.Right, otherwise Dlarfx will panic.
+//
+// c and ldc represent the m×n matrix C. On return, C is overwritten by the
+// matrix H * C if side == blas.Left, or C * H if side == blas.Right.
+//
+// work must have length at least n if side == blas.Left, and at least m if side
+// == blas.Right, otherwise Dlarfx will panic. work is not referenced if H has
+// order < 11.
+func (impl Implementation) Dlarfx(side blas.Side, m, n int, v []float64, tau float64, c []float64, ldc int, work []float64) {
+	checkMatrix(m, n, c, ldc)
+	switch side {
+	case blas.Left:
+		checkVector(m, v, 1)
+		if len(work) < n && m > 10 {
+			panic(badWork)
+		}
+	case blas.Right:
+		checkVector(n, v, 1)
+		if len(work) < m && n > 10 {
+			panic(badWork)
+		}
+	default:
+		panic(badSide)
+	}
+
+	if tau == 0 {
+		return
+	}
+
+	if side == blas.Left {
+		// Form H * C, where H has order m.
+		switch m {
+		default:
+			impl.Dlarf(side, m, n, v, 1, tau, c, ldc, work)
+		case 0:
```

Comment:
Maybe add a comment on the m == 0 case.

I think given the size of the case blocks, judicious addition of new lines after the returns is warranted. I looks nicer in this case I think.